localstack/services/apigateway/helpers.py

```diff
@@ -1186,42 +1261,65 @@ def add_path(path, parts, parent_id=""):
             path_part=parts[-1] or "/",
             parent_id=parent_id,
         )
-        for m, payload in body["paths"].get(path, {}).items():
-            m = m.upper()
-            payload = payload["x-amazon-apigateway-integration"]
 
-            child.add_method(m, None, None)
+        for method, method_schema in resolved_schema["paths"].get(path, {}).items():
+            method = method.upper()
+            method_resource = child.add_method(method, None, None)
+            method_integration = method_schema.get("x-amazon-apigateway-integration", {})
+            responses = method_schema.get("responses", {})
+            for status_code in responses:
+                response_model = None
+                if model_schema := responses.get(status_code, {}).get("schema", {}):
+                    response_model = {APPLICATION_JSON: model_schema}
+
+                response_parameters = (
+                    method_integration.get("responses", {})
+                    .get("default", {})
+                    .get("responseParameters")
+                )
+                method_resource.create_response(
+                    status_code,
+                    response_model,
+                    response_parameters,
+                )
+
             integration = apigateway_models.Integration(
-                http_method=m,
-                uri=payload.get("uri"),
-                integration_type=payload["type"],
-                pass_through_behavior=payload.get("passthroughBehavior"),
-                request_templates=payload.get("requestTemplates") or {},
+                http_method=method,
+                uri=method_integration.get("uri"),
+                integration_type=method_integration["type"],
+                passthrough_behavior=method_integration.get("passthroughBehavior"),
+                request_templates=method_integration.get("requestTemplates") or {},
             )
             integration.create_integration_response(
-                status_code=payload.get("responses", {}).get("default", {}).get("statusCode", 200),
+                status_code=method_integration.get("default", {}).get("statusCode", 200),
                 selection_pattern=None,
-                response_templates=None,
+                response_templates=method_integration.get("default", {}).get(
+                    "responseTemplates", None
+                ),
                 content_handling=None,
             )
-            child.resource_methods[m]["methodIntegration"] = integration
+            child.resource_methods[method]["methodIntegration"] = integration
 
         rest_api.resources[child_id] = child
         return child
 
+    if definitions := resolved_schema.get("definitions"):
```

Comment:
```suggestion
    if definitions := resolved_schema.get("definitions", {}):
```

Reply:
Better hedge against the case where `"definitions"` is potentially not present.

Reply:
We established an agreement, that is not required.

Reply:
Yep, this comment can be ignored. Totally missed the guard on line 1306 - thanks for pointing it out. 👍